salt/hosts-master/init.sls

```diff
@@ -0,0 +1,26 @@
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-minion', 'network.ip_addrs', 'grain').items() %}
+{{ minion_id }}-host-entry:
+  host.present:
+{% if addrlist is string %}
+    - ip: {{ addrlist }}
+{% else %}
+    - ip: {{ addrlist|first }}
+{% endif %}
+    - names:
+      - {{ minion_id }}
+      - {{ minion_id }}.{{ pillar['internal_infra_domain'] }}
+{% endfor %}
+{%- for minion_id, addrlist in salt['mine.get']('roles:kube-master', 'network.ip_addrs', 'grain').items() %}
```

Comment:
We will need each master to know about the other masters - I can swap the api to 127.0.0.1